crates/holochain_integrity_types/src/chain.rs

```diff
@@ -16,87 +16,96 @@ mod test;
 /// Filter source chain items.
 /// Starting from some chain position given as an [`ActionHash`]
 /// the chain is walked backwards to genesis.
-/// The filter can stop early by specifying the number of
-/// chain items to take and / or an [`ActionHash`] to consume until.
+/// The filter can stop early by specifying stop conditions:
+/// A maximum number of items is reached, a given [`ActionHash`] is found, or
+/// a given timestamp has been passed.
+/// Multiple `[`ActionHash`] can be provided. The filter will stop at and take the first one found.
+/// When providing a Timestamp, the filter will stop at and take the last item that has a
+/// Timestamp younger than the provided one. In the edge case of multiple actions having the same
+/// Timestamp as the stop condition, only one of the actions will be included.
```

Comment:
This has to be more specific. It sounds like it could be either but it either needs to be well-defined as the one with the highest sequence number or we need to re-examine this logic and we should be including actions past the provided bound.

I'm inclined to think it's the latter. If you provide a timestamp bound, surely all timestamps within that bound should be included and there is no edge case here?